src/astx/flows.py

```diff
@@ -418,3 +418,104 @@ def get_struct(self, simplified: bool = False) -> ReprStruct:
         }
 
         return self._prepare_struct(key, value, simplified)
+
+
+@public
+@typechecked
+class CaseStmt(StatementType):
+    """AST class for a branch in a switch/match expression."""
+
+    condition: Optional[Expr] = None
+    body: Expr
+    default: bool = False
+
+    def __init__(
+        self,
+        body: Expr,
+        condition: Optional[Expr] = None,
+        default: bool = False,
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+        parent: Optional[ASTNodes] = None,
+    ) -> None:
+        """Initialize the CaseStmt instance."""
+        super().__init__(loc=loc, parent=parent)
+        self.condition = condition
+        self.body = body
+        self.default = default
+        self.kind = ASTKind.CaseStmtKind
+
+        if self.default is False and self.condition is None:
+            raise ValueError(
+                "Condition must be provided for non-default branches."
+            )
+
+        if self.default is True and self.condition is not None:
+            raise ValueError(
+                "Condition must NOT be provided for default branches."
+            )
+
+    def __str__(self) -> str:
+        """Return a string representation of the object."""
+        return f"CaseStmt[{self.condition} => {self.body}]"
```

Comment:
it is not necessary to show the body, if the users want to check the body, they can use get_structs (for example)

Reply:
but it should return the condition if it is not the default branch, otherwise, it should return "default"